task-sdk/src/airflow/sdk/execution_time/task_runner.py

```diff
@@ -677,6 +685,34 @@ def startup() -> tuple[RuntimeTaskInstance, Context, Logger]:
         ti.log_url = get_log_url_from_ti(ti)
     log.debug("DAG file parsed", file=msg.dag_rel_path)
 
+    try:
+        run_as_user = getattr(ti.task, "run_as_user", None) or conf.get("core", "default_impersonation")
```

Comment:
The other thing Airflow 2 did here was check that run_as_user was actually different to the current user.

We should probably do that as well.